bee-common/src/shutdown.rs

```diff
@@ -0,0 +1,83 @@
+// Copyright 2020 IOTA Stiftung
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+// the License. You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+// an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and limitations under the License.
+
+use crate::worker::Error as WorkerError;
+
+use futures::channel::oneshot;
+use log::error;
+use thiserror::Error;
+
+use std::future::Future;
+
+#[derive(Error, Debug)]
+pub enum Error {
+    #[error("Sending the shutdown signal to a worker failed.")]
+    SendingShutdownSignalFailed,
+
+    #[error("Waiting for worker to shut down failed.")]
+    WaitingforWorkerShutdownFailed(#[from] WorkerError),
+}
+
+pub type ShutdownNotifier = oneshot::Sender<()>;
+pub type ShutdownListener = oneshot::Receiver<()>;
+pub type WorkerShutdown = Box<dyn Future<Output = Result<(), WorkerError>> + Unpin>;
+pub type Action = Box<dyn Fn()>;
```

Comment:
Can you add documentation ? You have them in the RFC already :)

Reply:
yup, will do